pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/extension/PythonDetails.java

```diff
@@ -114,15 +116,16 @@ public void appendExecutableDirectory(File file) {
         searchPath.add(file);
     }
 
-    public void setPythonVersion(String pythonVersion) {
-        if ("2".equals(pythonVersion)) {
-            pythonVersion = "2.6";
-        }
+    public void setPythonDefaults(PythonDefaults defaults) {
+        pythonDefaults = defaults;
+    }
 
-        if ("3".equals(pythonVersion)) {
-            pythonVersion = "3.5";
-        }
+    public PythonDefaults getPythonDefaults() {
+        return pythonDefaults;
+    }
 
+    public void setPythonVersion(String version) {
+        pythonVersion = new PythonVersion(pythonDefaults.normalize(version));
```

Comment:
If a user chooses a version that we as developers of PyGradle do not support (e.g., 3.3), I believe the user should also set `allowedVersions` if they think it will work anyway for them. So, I'm fine with Barry's change here.